features/admin.remote-userstores.v1/components/create/configurations-form.tsx

```diff
@@ -93,6 +95,11 @@ const ConfigurationsForm: ForwardRefExoticComponent<RefAttributes<Configurations
 
         const { t } = useTranslation();
 
+        const userStoreFeatureConfig: FeatureAccessConfigInterface = useSelector((state: AppState) =>
```

Comment:
Shall we move this inside the `getReadGroupFeatureEnabled` function?

Reply:
Ideally, the util function should accept these as parameters and calculate result, right? The same pattern we have followed for `useRequiredScopes`, `isFeatureEnabled` utils.

Reply:
Functions such as useRequiredScope etc are generic, hence it makes sense to parameterize the feature config.

In this particular case, this function only consumes the userstore config, so it'd be better to move it inside IMO.

Reply:
- IMO util function should not interact with the state rather component should. Isn't the React best practice; Separation of concerns?
"A utility function should focus on logic, not on interacting with the state"
- In case we are writing unit tests for the util function, we can write with mock inputs when it's implemented this way. Otherwise we'll have to mock the state.

Reply:
> "A utility function should focus on logic, not on interacting with the state"

Yes, only when it's generic, but in this case, it's not, is it?

> In case we are writing unit tests for the util function, we can write with mock inputs when it's implemented this way. Otherwise we'll have to mock the state.

Either way, we have to mock the state to get the feature config :) 

@brionmario WDYT?

Reply:
Shall we move this to a hook `useReadGroupsFeatureStatus` or something similar.
Reason is, since we are planning to move away from `Redux` to `Context API`, a hook will make that process easier than a util since we can't directly access the Context API store like we do with `Redux`.